### PR TITLE
[system-probe] Fix DNS cache name lost

### PR DESCRIPTION
**Pull request by oiooj**
### What does this PR do?

Fix DNS cache merge bug

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: pkg/ebpf/dns_cache.go [hunk 1]
```diff
@@ -185,12 +184,13 @@ type dnsCacheVal struct {
 }
 
 func (v *dnsCacheVal) merge(name string) {
-	if sort.SearchStrings(v.names, name) < len(v.names) {
```
Comment (p-lambert) [inline pkg/ebpf/dns_cache.go]: Can we preserve the binary search instead? The fixed conditional would be:
```
if i := sort.SearchStrings(v.names, name); i < len(v.names) && v.names[i] == name {
```

File: pkg/ebpf/dns_cache_test.go [hunk 2]
```diff
@@ -169,6 +169,32 @@ func TestDNSCacheTelemetry(t *testing.T) {
...
+	t1 := newTranslation([]byte("host-b"))
+	t1.add(util.AddressFromString("192.168.0.1"))
+	cache.Add(t1, ts)
+	res := cache.Get(conns, ts)
+	assert.Equal(t, []string{"host-b"}, res[util.AddressFromString("192.168.0.1")])
+
+	t2 := newTranslation([]byte("host-a"))
+	t2.add(util.AddressFromString("192.168.0.1"))
+	cache.Add(t2, ts)
+	res = cache.Get(conns, ts)
```
Comment (p-lambert) [inline pkg/ebpf/dns_cache_test.go]: Could we also add a `t3` re-adding the same translation to make sure the result won't contain duplicate entries?

**Review by p-lambert — COMMENTED** [review]
Thank you very much for your contribution! Left 2 comments.

File: pkg/ebpf/dns_cache_test.go (hunk 2 above)
Comment (oiooj) [reply pkg/ebpf/dns_cache_test.go]: Done

**Review by p-lambert — APPROVED** [review]

**Pull request merged**